docs/ui/auth/fragments/react/handleauthstatechange.md

```diff
@@ -0,0 +1,22 @@
+### handleAuthStateChange
+
+Similar to `onAuthUIStateChange`, `handleAuthStateChange` is a function you can attach to UI components that will fire whenever the state of the UI component changes.
+
+**Usage**
+
+```js
+import { AmplifyAuthenticator } from "@aws-amplify/ui-react";
+
+handleAuthStateChange((nextAuthState, authData) => {
+  if (nextAuthState === AuthState.SignedIn) {
+    console.log("user successfully signed in!");
+    console.log("user data: ", authData);
+  }
+  if (!authData) {
```

Comment:
Is it ever possible for `authData` to be falsey, but `nextAuthState` to be `AuthState.SignedIn`? If so, this might print both "user successfully signed in!" and "user is not signed in...".

```suggestion
  } else {
```

Reply:
Not sure, but I updated the code to just print out the data if it exists.